Felienne/hedy
# [BUG FIX] Users are able once again to set a favorite program from their programs page

**Description**
Due to a bug users where unable to set a favorite program from their programs page if they did have a public profile but currently not a favorite program selected. We fix this in this PR by slightly changing the db function on updating and return the users an extra error message in the unlikely case they try to set a favorite program without being logged in.

**Fixes**
This PR fixes #2213

**How to test**
Make sure you have a public profile **without** a favorite program. Verify that you can set a favorite program from within the programs page by using the star symbol.
